x/oracle/keeper/query_native_token.go

```diff
@@ -0,0 +1,36 @@
+package keeper
+
+import (
+	"context"
+	"errors"
+
+	assetstypes "github.com/ExocoreNetwork/exocore/x/assets/types"
+	"github.com/ExocoreNetwork/exocore/x/oracle/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) StakerInfos(goCtx context.Context, req *types.QueryStakerInfosRequest) (*types.QueryStakerInfosResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request for stakerInfos")
+	}
+	if !assetstypes.IsNativeToken(req.AssetID) {
+		return nil, errors.New("assetID doesn't reprensents any supported nativeRestakingToken")
```

Comment:
the error can be var like the following:
>var (
    ErrInvalidRequest = status.Error(codes.InvalidArgument, "invalid request")
    ErrUnsupportedAsset = errors.New("assetID doesn't represent any supported native restaking token")
)

Reply:
updated